go-controller/pkg/node/routemanager/route_manager.go

```diff
@@ -345,8 +345,50 @@ type RoutesPerLink struct {
 	Routes []Route
 }
 
-func (rl RoutesPerLink) Equal(rpl2 RoutesPerLink) bool {
-	return reflect.DeepEqual(rl, rpl2)
+func (rl RoutesPerLink) Equal(rl2 RoutesPerLink) bool {
```

Comment:
would be nice in the commit message to describe how they were not being compared correctly with deep equal